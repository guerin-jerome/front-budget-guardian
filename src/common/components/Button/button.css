button {
  height: 38px;
  width: inherit;
  font-size: 16px;
  font-family: Arial, sans-serif;
  border: 0 solid;
  border-radius: 5px;
  cursor: pointer;
  color: white;
}

button:hover {
  transform: scale(1.015);
}

button.icon {
  width: 38px;
}

button.success {
  background-color: #218c74;
}

button.danger {
  background-color: #b33939;
}

button.primary {
  background-color: #0077c2;
}

button.secondary {
  background-color: #333;
}
