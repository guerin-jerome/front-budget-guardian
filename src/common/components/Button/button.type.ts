import { MouseEventHandler, ReactNode } from "react";

export enum ButtonAppearence {
  PRIMARY = "primary",
  SECONDARY = "secondary",
  DANGER = "danger",
  SUCCESS = "success",
}

export type ButtonProps = {
  onClick: MouseEventHandler;
  children: ReactNode;
  type?: "button" | "submit" | "reset";
  appearence?: ButtonAppearence;
  className?: string;
  isIconMode?: boolean;
};
