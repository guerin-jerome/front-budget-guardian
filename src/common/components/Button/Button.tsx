import { ButtonAppearence, ButtonProps } from "./button.type";
import classNames from "classnames";
import "./button.css";

export const Button = ({
  children,
  className,
  isIconMode,
  type = "button",
  appearence = ButtonAppearence.PRIMARY,
  onClick = () => {},
}: ButtonProps) => (
  <button
    className={classNames(className, appearence, { icon: isIconMode })}
    onClick={onClick}
    type={type}
  >
    {children}
  </button>
);
