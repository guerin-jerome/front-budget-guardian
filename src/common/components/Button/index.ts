import { Button } from "./Button";
import { ButtonAppearence } from "./button.type";

export { Button, ButtonAppearence };
