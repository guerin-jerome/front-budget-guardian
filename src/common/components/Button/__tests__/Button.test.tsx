import { render, screen } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";
import { Button } from "../Button";
import { ButtonAppearence } from "../button.type";
import userEvent from "@testing-library/user-event";

describe("<Button />", () => {
  const onClick = vi.fn();

  it("matchs snapshot - cas nominal", () => {
    const { container } = render(<Button onClick={onClick}>Hello</Button>);
    expect(container).toMatchSnapshot();
  });

  it("renders well - secondary className iconMode", async () => {
    render(
      <Button
        onClick={onClick}
        isIconMode
        appearence={ButtonAppearence.SECONDARY}
        className="test"
      >
        Hello
      </Button>
    );

    const button = screen.getByRole("button", { name: "Hello" });

    await userEvent.click(button);

    expect(button).toBeDefined();
    expect(onClick).toHaveBeenCalled();
  });

  it("renders well - secondary submit", async () => {
    render(
      <Button
        onClick={onClick}
        type="submit"
        appearence={ButtonAppearence.SECONDARY}
      >
        Hello
      </Button>
    );

    const button = screen.getByRole("button", { name: "Hello" });

    expect(button).toBeDefined();
  });

  it("renders well - tertiary reset", async () => {
    render(
      <Button
        onClick={onClick}
        type="reset"
        appearence={ButtonAppearence.TERTIARY}
      >
        Hello
      </Button>
    );

    const button = screen.getByRole("button", { name: "Hello" });

    expect(button).toBeDefined();
  });
});
