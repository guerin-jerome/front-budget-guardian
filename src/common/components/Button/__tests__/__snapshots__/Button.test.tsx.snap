// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Button /> > matchs snapshot - cas nominal 1`] = `
<div>
  <button
    class="primary"
    type="button"
  >
    Hello
  </button>
</div>
`;
